### ISSUE TITLE
Prettier plugins ignored when ESLint parser is not parserPlain

### ISSUE THREAD

**Issue by gtbuchanan** [antfu/eslint-plugin-format#21]
## Description

When the ESLint parser for a file is anything other than `format.parserPlain` (e.g., `jsonc-eslint-parser` from another plugin), the `format/prettier` rule silently ignores all Prettier plugins passed in the rule options. The formatting runs but plugins don't load — no errors are reported.

Additionally, even with `parserPlain`, the synckit worker appears to cache Prettier's plugin state from the first invocation. When different file patterns use `format/prettier` with different plugin sets (e.g., `sort-json` for `*.json` and `multiline-arrays` for `package.json`), processing them together causes "Failed to format the code" errors on whichever pattern is processed second.

### Bug 1: Plugins ignored without parserPlain

```js
import format from 'eslint-plugin-format';
import { configs as pnpmConfigs } from 'eslint-plugin-pnpm';

export default [
  // pnpm plugin sets jsonc-eslint-parser on package.json
  ...pnpmConfigs.json,
  {
    files: ['**/package.json'],
    plugins: { format },
    rules: {
      'format/prettier': ['warn', {
        parser: 'json',
        plugins: ['prettier-plugin-multiline-arrays'],
        multilineArraysWrapThreshold: 0,
      }],
    },
  },
];
```

**Expected:** `prettier-plugin-multiline-arrays` forces all arrays to one-element-per-line.
**Actual:** The plugin is silently ignored.

### Bug 2: Worker state corruption across file patterns

```js
import format from 'eslint-plugin-format';

export default [
  {
    files: ['**/*.json'],
    ignores: ['**/package.json'],
    languageOptions: { parser: format.parserPlain },
    plugins: { format },
    rules: {
      'format/prettier': ['warn', {
        parser: 'json',
        plugins: ['prettier-plugin-sort-json', 'prettier-plugin-multiline-arrays'],
      }],
    },
  },
  {
    files: ['**/package.json'],
    languageOptions: { parser: format.parserPlain },
    plugins: { format },
    rules: {
      'format/prettier': ['warn', {
        parser: 'json',
        plugins: ['prettier-plugin-multiline-arrays'],
      }],
    },
  },
];
```

Running `eslint package.json` alone works. Running `eslint package.json tsconfig.json` (both match different configs above) causes "Failed to format the code" on `package.json`.

## Workaround

No complete workaround found. Using identical plugin sets across all `format/prettier` configs avoids bug 2, but bug 1 prevents using plugins at all when other ESLint plugins (like `eslint-plugin-pnpm`) set a non-`parserPlain` parser on the same files.

## Environment

- eslint-plugin-format: 2.0.1
- eslint: 10.1.0
- prettier: 3.8.3
- Node.js: 24.x

**Comment by gtbuchanan** [comment]
Closing — the root cause was on my end. I was using `parser: 'json'` for `package.json` files when I should have been using `parser: 'json-stringify'` (registered by `prettier-plugin-packagejson`). With the correct parser, plugins load and compose correctly without needing `parserPlain`, and other ESLint plugins that need a JSON AST (like `eslint-plugin-pnpm`) continue to work.

The "worker state corruption across file patterns" (Bug 2 in my report) was also a consequence of the wrong parser — with `json-stringify`, different file patterns with different plugin sets work correctly in the same ESLint run.

Apologies for the noise.
